### PR TITLE
More faster

### PR DESCRIPTION
**Pull request by dgreisen-cfpb**
this also contains all of #74. switch over most of universalclient to urllib3. also geventize git calls. 1/3 faster with no caching (45 -> 30 min). 50% faster with caching (30 -> 15 min).

### CODE REVIEW

File: server/indexers/gh_issues.py [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
 
 
-def is_updated_issues(client, repo_name, since):
-    issues = client.repos._(repo_name).issues.params(
-        state='all', since=since)
-    return True if issues.get().json() else False
+def is_updated_issues(gh_type, pool, repo_name, since):
+    fields = {'state': 'all'}
+    if since:
+        fields = {'state': 'all', 'since': since}
```
Comment (m3brown) [inline server/indexers/gh_issues.py]: Nitpick, this is redundant.  `fields['since'] = since` would be cleaner.

File: server/indexers/ghe.py [hunk 2]
```diff
@@ -10,18 +10,24 @@
...
+                                           headers=urllib3.util.make_headers(keep_alive=True)
+                                          )
 ghe_pool = urllib3.connection_from_url(ghe_settings.get('WEB'), maxsize=50, block=True)
 
-def _get_repos():
+def get_repos():
     if not ghe_settings:
         return []
-    return [repo['full_name'] for repo in utils.iter_get(ghe_api_client.repositories) if not repo['fork']]
+    return [repo['full_name'] for repo in utils.iter_get_url('/api/v3/repositories', ghe_api_pool) if not repo['fork']]
```
Comment (m3brown) [inline server/indexers/ghe.py]: This is a good place to use the new settings.API_PATH

File: server/indexers/gh.py [hunk 3]
```diff
@@ -10,15 +10,24 @@
 from gh_pages import index as gh_pages
 from gh_issues import index as gh_issues
 
-gh_pool = urllib3.connection_from_url(settings.GITHUB.get('GH', {}).get('WEB'), maxsize=50, block=True)
+headers = {
+    'keep_alive': True,
+    'user_agent': 'cfpb-tiresias',
+}
+
 gh_api_client = Client(gh_settings.get('API'))
```
Comment (m3brown) [inline server/indexers/gh.py]: It looks like all the universal client code is not being used (except for jira's process).  You should probably pull out all the unused instances.

Comment (dgreisen-cfpb) [inline server/indexers/gh.py]: it's still used in the tests. The tests will need to be cleaned up before the uc instances can be removed.

they could be moved to the test code, though

Comment (dgreisen-cfpb) [inline server/indexers/gh.py]: actually, I think this should be another pr. Should just go through and remove all of UC, and make sure nothing will be broken by grepping grough everything. I don't think it makes sense to risk breaking something 2.5 hours before review.
